### ISSUE TITLE
[Feature] Set window class

### ISSUE THREAD

**Issue by SpaceboyRoss01** [sidorares/ntk#31]
Can we please have support for all of the EWMH stuff besides the title?

**Comment by sidorares** [comment]
yes. Do you want to volunteer? How can I help you start?
https://github.com/santigimeno/node-ewmh

**Comment by SpaceboyRoss01** [comment]
Yeah, I'll volunteer. I'm pretty good at figuring out how to "reverse engineer" code so I don't really need any help.

**Comment by sidorares** [comment]
Still open as of the current codebase. Status today:

- `lib/window.js` sets `WM_NAME` via `setTitle()` and opts into `WM_PROTOCOLS`/`WM_DELETE_WINDOW` via `setActions()` — that's the extent of ICCCM/EWMH property support.
- No `WM_CLASS` setter and no `_NET_WM_*` properties (`_NET_WM_NAME`, `_NET_WM_STATE`, `_NET_WM_WINDOW_TYPE`, `_NET_WM_ICON`, ...) yet.

Implementation notes: `WM_CLASS` is strictly ICCCM (predefined atom 67, already exposed by node-x11's `stdatoms`), so a `setClass(instance, cls)` is a small `ChangeProperty` call next to `setTitle()` in `lib/window.js` — two NUL-terminated strings. Broader EWMH support (window type, state, icons) would fit as a handful of similar helpers there; https://github.com/santigimeno/node-ewmh is still a useful reference for the property formats.

PRs welcome.

**Comment by sidorares** [comment]
Both halves of this are done — the window class, and the "all of the EWMH stuff besides the title" that the body actually asks for.

`setClass(instance, class)` writes `WM_CLASS` as the two NUL-terminated latin-1 strings taskbars group by, with a `wmClass` creation argument. Verified on master (`7621477`) by reading the property back over the wire: `"demo\0Demo\0"`.

EWMH beyond the title, as of #130/#131/#132:

- `setWindowType(type)` — `_NET_WM_WINDOW_TYPE`, short names expanded, fallback lists supported
- `setWmState(names, action)` / `getWmStates()` / `statechange` — `_NET_WM_STATE`: fullscreen, maximized, sticky, skip-taskbar, demands-attention and the rest. Mapped windows ask the window manager, unmapped ones declare the state by writing the property, per EWMH 7.7
- `setPid()` — `_NET_WM_PID` + `WM_CLIENT_MACHINE`, automatic on top-level windows
- `setTransientFor(owner)`, `setWmHints({...})`, `setProtocols(...)` — the ICCCM side that EWMH sits on
- `getProperty` / `setProperty` / `deleteProperty` for anything not wrapped

Docs: [Window manager hints](https://github.com/sidorares/ntk/blob/master/docs/window.md#window-manager-hints).

Still not written: `_NET_WM_ICON`, tracked in #118. That is a pixel-format decision rather than a missing writer — EWMH wants non-premultiplied ARGB packed in CARD32s, ntk's `getImageData` returns BGRA bytes, and canvas `ImageData` is RGBA — so it wants its own change with a convention chosen deliberately.

Closing; #118 carries the remainder.
